apartments/tests.py

```diff
@@ -76,3 +78,61 @@ def test_update_apartment_view_to_not_owner_user(self, client, new_user):
         assert response.status_code == 302
         response = client.get(response.url)
         assert response.status_code == 200
+
+
+@pytest.mark.parametrize(
+    'city, address, rent, num_of_roomates, num_of_rooms, start_date, about, validity',
+    [
+        ('new_city', 'address', 1000, 2, 2, '2021-2-2', 'about', True),
+        ('new_city', '', 1000, 2, 2, '2021-2-2', 'about', False),
+        ('new_city', 'address', None, 2, 2, '2021-2-2', 'about', False),
+        ('new_city', 'address', 1000, None, 2, '2021-2-2', 'about', False),
+        ('new_city', 'address', 1000, 2, None, '2021-2-2', 'about', False),
+        ('new_city', 'address', 1000, 2, 2, '', 'about', False),
+        ('new_city', 'address', 1000, 2, 2, '2021-2-2', '', True),
+        ('new_city', 'address', -1, 2, 2, '2021-2-2', 'about', False),
+        ('new_city', 'address', 1000, -1, 2, '2021-2-2', 'about', False),
+        ('new_city', 'address', 1000, 2, -1, '2021-2-2', 'about', False),
+    ])
+@pytest.mark.django_db
+def test_apartment_form_validity(city, address, rent, num_of_roomates, num_of_rooms, start_date, about,
+                                 validity, request):
+
+    city = request.getfixturevalue(city)
+    form = ApartmentCreationForm(data={
+        'city': city,
+        'address': address,
+        'rent': rent,
+        'num_of_roomates': num_of_roomates,
+        'num_of_rooms': num_of_rooms,
+        'start_date': start_date,
+        'about': about,
+    })
+
+    assert form.is_valid() is validity
+
+
+@pytest.fixture
+def valid_form(new_city):
+    return ApartmentCreationForm(data={
```

Comment:
shouldn't you use `form.is_valid()`?

Reply:
It's a fixture I later use in `test_fail_to_save_apartment_form_with_commit_true`, do you mean that I should assert that the form is really valid in the fixture?

Reply:
Sorry, I didn't notice It's the fixture itself. 
Do not change it, but do add a separate test that asserts an insertion of a valid form was successful.

Reply:
Will do